dhis-2/dhis-web/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/MvcTestAuthConfig.java

```diff
@@ -0,0 +1,205 @@
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.web.accept.ContentNegotiationManager;
+import org.springframework.web.accept.FixedContentNegotiationStrategy;
+import org.springframework.web.accept.HeaderContentNegotiationStrategy;
+import org.springframework.web.accept.ParameterContentNegotiationStrategy;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.servlet.config.annotation.DelegatingWebMvcConfiguration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+
+import com.google.common.collect.ImmutableMap;
+
+/**
+ * @author Morten Svanæs <msvanaes@dhis2.org>
+ */
+@Configuration
+@EnableWebMvc
+@EnableGlobalMethodSecurity( prePostEnabled = true )
+public class MvcTestAuthConfig extends DelegatingWebMvcConfiguration
+{
+    @Autowired
+    private CurrentUserService currentUserService;
+
+    @Autowired
+    private UserSettingService userSettingService;
+
+    @Autowired
+    private CurrentUserHandlerMethodArgumentResolver currentUserHandlerMethodArgumentResolver;
+
+    @Bean
+    public CustomRequestMappingHandlerMapping requestMappingHandlerMapping()
+    {
+        CustomPathExtensionContentNegotiationStrategy pathExtensionNegotiationStrategy = new CustomPathExtensionContentNegotiationStrategy(
+            mediaTypeMap );
+        pathExtensionNegotiationStrategy.setUseJaf( false );
+
+        String[] mediaTypes = new String[] { "json", "jsonp", "xml", "png", "xls", "pdf", "csv" };
+
+        ParameterContentNegotiationStrategy parameterContentNegotiationStrategy = new ParameterContentNegotiationStrategy(
+            mediaTypeMap.entrySet().stream()
+                .filter( x -> ArrayUtils.contains( mediaTypes, x.getKey() ) )
+                .collect( Collectors.toMap( Map.Entry::getKey, Map.Entry::getValue ) ) );
+
+        HeaderContentNegotiationStrategy headerContentNegotiationStrategy = new HeaderContentNegotiationStrategy();
+        FixedContentNegotiationStrategy fixedContentNegotiationStrategy = new FixedContentNegotiationStrategy(
+            MediaType.APPLICATION_JSON );
+
+        ContentNegotiationManager manager = new ContentNegotiationManager(
+            Arrays.asList(
+                pathExtensionNegotiationStrategy,
+                parameterContentNegotiationStrategy,
+                headerContentNegotiationStrategy,
+                fixedContentNegotiationStrategy ) );
+
+        CustomRequestMappingHandlerMapping mapping = new CustomRequestMappingHandlerMapping();
+        mapping.setOrder( 0 );
+        mapping.setContentNegotiationManager( manager );
+
+        return mapping;
+    }
+
+    private Map<String, MediaType> mediaTypeMap = new ImmutableMap.Builder<String, MediaType>()
```

Comment:
Potential for externalizing and re-using in other locations?

Reply:
Ah, this is a dead class. I removed it.